### PR TITLE
Babel setup to prepare for android

### PR DESCRIPTION
**Pull request by anderklander**
Also added the separate app.js file for android.

Please verify function on Linux and macOS!

---
This change is [Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/44)

### CODE REVIEW

**Comment by pronebird** [comment]
Review status: 0 of 5 files reviewed at latest revision, 1 unresolved discussion.

---

*[package.json, line 22 at r1](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44#-L3coPojPMJWUBLgjsIt:-L3coPojPMJWUBLgjsIu:b5bt2lo) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/40efbb790f5776fad7336c4ba19db384179511f4/package.json#L22)):*
> ```JSON
>     "mkdirp": "^0.5.1",
>     "moment": "^2.17.1",
>     "prop-types": "^15.5.10",
> ```

Good one, we actually don't use it because we have Flow. :)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44)*

**Comment by pronebird** [comment]
Reviewed 1 of 5 files at r1.
Review status: 1 of 5 files reviewed at latest revision, 1 unresolved discussion.

---

*[app/app.android.js, line 25 at r1](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44#-L3cpPz5kK1WLuV7GOlU:-L3cpPz5kK1WLuV7GOlV:b44dvzc) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/75b3202ff3ee48fec418402bd7a63c0d46d674a4/app/app.android.js#L25)):*
> ```JavaScript
> 
> DeviceEventEmitter.addListener('com.mullvad.backend-info', function(e: Event) {
>   console.warn(e);
> ```

`console.warn`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44)*

**Comment by pronebird** [comment]
Review status: 1 of 5 files reviewed at latest revision, 2 unresolved discussions.

---

*[app/app.android.js, line 56 at r1](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44#-L3cpdUuvzJ9RqGRkpEp:-L3cpdUuvzJ9RqGRkpEq:b-8g4h24) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/75b3202ff3ee48fec418402bd7a63c0d46d674a4/app/app.android.js#L56)):*
> ```JavaScript
> 
>   componentWillMount() {
> ```

Dead code

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44)*

**Comment by pronebird** [comment]
Reviewed 4 of 5 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44)*

**Comment by pronebird** [comment]
Thanks! Super easy to review when it come in such size!

---

Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44#-:-L3cphPXxZ033jHX0_Sr:b-a4me1o)*

**Comment by pronebird** [comment]
Review status: 4 of 5 files reviewed at latest revision, 2 unresolved discussions.

---

*[.babelrc, line 7 at r2](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44#-L3crHM7-Rb4VaKoCZb2:-L3crHM7-Rb4VaKoCZb3:b-mqpgae) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/c8aa459bc725e60bcb0210643f99d23c2f0a8fed/.babelrc#L7)):*
> ```JSON5
>       "plugins": ["transform-runtime", "inline-react-svg"]
>     },
>     "development": {
> ```

I'd like this to be documented somewhere with the issue #8723 @ react-native reference:

https://github.com/facebook/react-native/issues/8723

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44)*

**Comment by anderklander** [comment]
Review status: 4 of 6 files reviewed at latest revision, 3 unresolved discussions.

---

*[.babelrc, line 7 at r2](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44#-L3crHM7-Rb4VaKoCZb2:-L3csd-kIYuRYXi98vrb:b-896fix) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/c8aa459bc725e60bcb0210643f99d23c2f0a8fed/.babelrc#L7)):*
<details><summary><i>Previously, pronebird (Andrei Mihailov) wrote…</i></summary><blockquote>

I'd like this to be documented somewhere with the issue #8723 @ react-native reference:

https://github.com/facebook/react-native/issues/8723
</blockquote></details>

Done.

---

*[app/app.android.js, line 25 at r1](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44#-L3cpPz5kK1WLuV7GOlU:-L3csZfEnib_ilE6Gplq:b-896fix) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/75b3202ff3ee48fec418402bd7a63c0d46d674a4/app/app.android.js#L25)):*
<details><summary><i>Previously, pronebird (Andrei Mihailov) wrote…</i></summary><blockquote>

`console.warn`
</blockquote></details>

Done.

---

*[app/app.android.js, line 56 at r1](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44#-L3cpdUuvzJ9RqGRkpEp:-L3cs_72RgjIQZJx89Wu:b-896fix) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/75b3202ff3ee48fec418402bd7a63c0d46d674a4/app/app.android.js#L56)):*
<details><summary><i>Previously, pronebird (Andrei Mihailov) wrote…</i></summary><blockquote>

Dead code
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44)*

**Comment by pronebird** [comment]
Reviewed 1 of 2 files at r3.
Review status: 5 of 6 files reviewed at latest revision, 3 unresolved discussions.

---

*[README.md, line 138 at r3](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44#-L3csfsyGwc34ACnSB94:-L3csfsyGwc34ACnSB95:b3n0ksr) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/aa80134d9e6bc1df3862d80caa26f067083d1d64/README.md#L138)):*
> ```Markdown
> ## Quirks
> 
> - `BABEL_ENV=development` [must be used for react-native](https://github.com/facebook/react-native/issues/8723)
> ```

As long as @faern approves I approve that too. Otherwise alternative would be to have QUIRKS document or something like that.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44)*

**Comment by pronebird** [comment]
Reviewed 1 of 2 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44)*

**Comment by anderklander** [comment]
Review status: 5 of 6 files reviewed at latest revision, 2 unresolved discussions.

---

*[package.json, line 78 at r3](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44#-L3ctDaA6rW3STzuvwkk:-L3cu_i4LHzSpZZFpbn6:b-896fix) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/aa80134d9e6bc1df3862d80caa26f067083d1d64/package.json#L78)):*
<details><summary><i>Previously, pronebird (Andrei Mihailov) wrote…</i></summary><blockquote>

We may need to add a `cross-env` here for all `pack:*` commands since they clean and build from scratch.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44)*

**Comment by pronebird** [comment]
Review status: 5 of 6 files reviewed at latest revision, 2 unresolved discussions.

---

*[package.json, line 77 at r4](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44#-L3cukw6NxOdkjRz-JMs:-L3cukw6NxOdkjRz-JMt:b-gd4lrt) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/9b2b3163f91d5a3cfdaf1bd7e9842ed3a4bee565/package.json#L77)):*
> ```JSON
>     "lint": "eslint --no-ignore scripts app test *.js",
>     "flow": "flow",
>     "pack": "cross-env BABEL_ENV=electron run-s private:clean private:compile private:build:all",
> ```

Unrelated to your PR but since the `pack:mac` already runs a custom script it would be better to just run all `pack:` commands in sequence:

```
"pack": "run-s pack:mac pack:win pack:linux"
```

and remove `private:build:all` script.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44)*

**Comment by pronebird** [comment]
Reviewed 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44)*

**Comment by faern** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[package.json, line 77 at r4](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44#-L3cukw6NxOdkjRz-JMs:-L3cyHzDHwov8scAO8z0:bfbpe3a) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/9b2b3163f91d5a3cfdaf1bd7e9842ed3a4bee565/package.json#L77)):*
<details><summary><i>Previously, pronebird (Andrei Mihailov) wrote…</i></summary><blockquote>

Unrelated to your PR but since the `pack:mac` already runs a custom script it would be better to just run all `pack:` commands in sequence:

```
"pack": "run-s pack:mac pack:win pack:linux"
```

and remove `private:build:all` script.
</blockquote></details>

I don't think we'll ever pack all platforms from one. So can probably be removed and force developers to specify which one they want to build instead.

---

*[README.md, line 138 at r3](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44#-L3csfsyGwc34ACnSB94:-L3cySMs2uHTg56ftY0w:bwki2zo) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/aa80134d9e6bc1df3862d80caa26f067083d1d64/README.md#L138)):*
<details><summary><i>Previously, pronebird (Andrei Mihailov) wrote…</i></summary><blockquote>

As long as @faern approves I approve that too. Otherwise alternative would be to have QUIRKS document or something like that.
</blockquote></details>

If it's something that must always be done then I think it should be in the main instruction flow, not at the end when people have already failed. Also maybe describe better who needs to set in where and when. Like:
```
x. Do blah
y. If you develop towards mobile then: `export BABEL_ENV=development`
z. blah blah
```

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44)*

**Comment by anderklander** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[README.md, line 138 at r3](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44#-L3csfsyGwc34ACnSB94:-L3czHfzAmescJUJ9ciE:b15gmou) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/aa80134d9e6bc1df3862d80caa26f067083d1d64/README.md#L138)):*
<details><summary><i>Previously, faern (Linus Färnstrand) wrote…</i></summary><blockquote>

If it's something that must always be done then I think it should be in the main instruction flow, not at the end when people have already failed. Also maybe describe better who needs to set in where and when. Like:
```
x. Do blah
y. If you develop towards mobile then: `export BABEL_ENV=development`
z. blah blah
```
</blockquote></details>

It's not something you have to do if you just are using the existing scripts. More a fact to consider if you were to add more scripts or if you have a reason to add another babel configuration. Probably only relevant when upgrading packages or adding support for another platform or something like that. React-native does not handle the variable, and the default must be reserved for react-native (and hence the variable must be set when building for electron, which is included in the scripts so no extra if you follow the rest of the instructions) to make it possible to build on android using the file/project structure we want to have.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44)*

**Comment by faern** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[README.md, line 138 at r3](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44#-L3csfsyGwc34ACnSB94:-L3d1NnUUtDIY8dd2nNY:b-onzkq4) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/aa80134d9e6bc1df3862d80caa26f067083d1d64/README.md#L138)):*
<details><summary><i>Previously, anderklander wrote…</i></summary><blockquote>

It's not something you have to do if you just are using the existing scripts. More a fact to consider if you were to add more scripts or if you have a reason to add another babel configuration. Probably only relevant when upgrading packages or adding support for another platform or something like that. React-native does not handle the variable, and the default must be reserved for react-native (and hence the variable must be set when building for electron, which is included in the scripts so no extra if you follow the rest of the instructions) to make it possible to build on android using the file/project structure we want to have.
</blockquote></details>

Oh. So it's not needed in order to start using or developing the app? Then I'm a bit skeptical towards having it here. Then it's more a reminder for people adding build scripts later, right?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44)*

**Comment by anderklander** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[README.md, line 138 at r3](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44#-L3csfsyGwc34ACnSB94:-L3d1v3YBFB-ujWL0x1B:b-y15iuy) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/aa80134d9e6bc1df3862d80caa26f067083d1d64/README.md#L138)):*
<details><summary><i>Previously, faern (Linus Färnstrand) wrote…</i></summary><blockquote>

Oh. So it's not needed in order to start using or developing the app? Then I'm a bit skeptical towards having it here. Then it's more a reminder for people adding build scripts later, right?
</blockquote></details>

Correct, it would be very good to just have it as a comment in the code, if it were not just a thing in (auto-modified) JSON-files …  

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44)*

**Comment by faern** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[README.md, line 138 at r3](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44#-L3csfsyGwc34ACnSB94:-L3d46hR1apF6WF8kg5p:benge7v) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/aa80134d9e6bc1df3862d80caa26f067083d1d64/README.md#L138)):*
<details><summary><i>Previously, anderklander wrote…</i></summary><blockquote>

Correct, it would be very good to just have it as a comment in the code, if it were not just a thing in (auto-modified) JSON-files …  
</blockquote></details>

Ok, yes! I won't stop you from adding it to the README if you feel it's the correct place for it. At least for now, we can move it later if our documentation grows and requires better structure.

But could we maybe add a short comment about when it's needed? At the moment all it says is
"Quirk: variable_name=value",
which is kind of hard to parse. I see there is a link, but it would be nice if it was possible to understand, without leaving the README, if this line applies to me at all or not.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44)*

**Comment by anderklander** [comment]
Review status: 4 of 6 files reviewed at latest revision, 2 unresolved discussions.

---

*[package.json, line 77 at r4](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44#-L3cukw6NxOdkjRz-JMs:-L3d8obFghb7J_FeV7PO:b-896fix) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/9b2b3163f91d5a3cfdaf1bd7e9842ed3a4bee565/package.json#L77)):*
<details><summary><i>Previously, faern (Linus Färnstrand) wrote…</i></summary><blockquote>

I don't think we'll ever pack all platforms from one. So can probably be removed and force developers to specify which one they want to build instead.
</blockquote></details>

Done.

---

*[README.md, line 138 at r3](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44#-L3csfsyGwc34ACnSB94:-L3d8F7S_AwnpmZR_MUu:bbs3xcb) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/aa80134d9e6bc1df3862d80caa26f067083d1d64/README.md#L138)):*
<details><summary><i>Previously, faern (Linus Färnstrand) wrote…</i></summary><blockquote>

Ok, yes! I won't stop you from adding it to the README if you feel it's the correct place for it. At least for now, we can move it later if our documentation grows and requires better structure.

But could we maybe add a short comment about when it's needed? At the moment all it says is
"Quirk: variable_name=value",
which is kind of hard to parse. I see there is a link, but it would be nice if it was possible to understand, without leaving the README, if this line applies to me at all or not.
</blockquote></details>

Just added "Babel configuration:", feels really hard to find the sweet spot here, since explaining in detail would be too much i think. Do you have any suggestion?  

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44)*

**Comment by pronebird** [comment]
Review status: 4 of 6 files reviewed at latest revision, 2 unresolved discussions.

---

*[README.md, line 138 at r3](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/44#-L3csfsyGwc34ACnSB94:-L3dAQ0xAKmOLhMCkgHr:bwy2k0i) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/aa80134d9e6bc1df3862d80caa26f067083d1d64/README.md#L138)):*
<details><summary><i>Previously, anderklander wrote…</i></summary><blockquote>

Just added "Babel configuration:", feels really hard to find the sweet spot here, since explaining in detail would be too much i think. Do you have any suggestion?  
</blockquote></details>

I'd actually describe that the react-native uses a fixed name for babel configuration so we that's why the "development" configuration is meant to be for `react-native`. [ISSUE-URL]

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/44)*

**Comment by pronebird** [comment]
Review status: 4 of 6 files reviewed at latest revision, 2 unresolved discussions.

---

*[package.json, line 77 at r4](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/44#-L3cukw6NxOdkjRz-JMs:-L3dBWpd3LETQ1HsLBtM:b-ucfknh) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/9b2b3163f91d5a3cfdaf1bd7e9842ed3a4bee565/package.json#L77)):*
<details><summary><i>Previously, anderklander wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Thanks!

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/44)*

**Comment by pronebird** [comment]
Reviewed 1 of 2 files at r5.
Review status: 5 of 6 files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/44)*

**Comment by faern** [comment]
Review status: 5 of 6 files reviewed at latest revision, 2 unresolved discussions.

---

*[README.md, line 138 at r3](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44#-L3csfsyGwc34ACnSB94:-L3dBTE2NWATrsTBImkE:bq25b14) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/aa80134d9e6bc1df3862d80caa26f067083d1d64/README.md#L138)):*
<details><summary><i>Previously, pronebird (Andrei Mihailov) wrote…</i></summary><blockquote>

I'd actually describe that the react-native uses a fixed name for babel configuration so we that's why the "development" configuration is meant to be for `react-native`. [ISSUE-URL]
</blockquote></details>

"If you need to change/add a build script for a react-native target, don't forget to set `BABEL_ENV=development`. [Here is why](https://github.com/facebook/react-native/issues/8723)"

Or something. Since I don't know the full scope of this I can't craft a perfect text.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44)*

**Comment by faern** [comment]
Review status: 5 of 6 files reviewed at latest revision, 2 unresolved discussions.

---

*[README.md, line 138 at r3](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44#-L3csfsyGwc34ACnSB94:-L3dBwaOl_c9y7fXcu99:b-pst06e) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/aa80134d9e6bc1df3862d80caa26f067083d1d64/README.md#L138)):*
<details><summary><i>Previously, faern (Linus Färnstrand) wrote…</i></summary><blockquote>

"If you need to change/add a build script for a react-native target, don't forget to set `BABEL_ENV=development`. [Here is why](https://github.com/facebook/react-native/issues/8723)"

Or something. Since I don't know the full scope of this I can't craft a perfect text.
</blockquote></details>

The most important is not exactly how to use it, but to quickly say who *don't* need to care about this. Since it's not relevant for most(?) users it's good if they can remove this section from their mental overhead as quickly as possible.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44)*

**Comment by faern** [comment]
Review status: 5 of 6 files reviewed at latest revision, 2 unresolved discussions.

---

*[README.md, line 138 at r3](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44#-L3csfsyGwc34ACnSB94:-L3dCOwodtIMxDTA_7Sd:bateuha) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/aa80134d9e6bc1df3862d80caa26f067083d1d64/README.md#L138)):*
<details><summary><i>Previously, faern (Linus Färnstrand) wrote…</i></summary><blockquote>

The most important is not exactly how to use it, but to quickly say who *don't* need to care about this. Since it's not relevant for most(?) users it's good if they can remove this section from their mental overhead as quickly as possible.
</blockquote></details>

Imagine someone like me cloning this repo and wanting to play with building and running the app. I don't even know what Babel is. I just want a series of commands to run in order to get the GUI running. Maybe I want to hack on the backend, but I don't know any JS/Babel/Node/Electron, I just want instructions that will help me get that part working.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44)*

**Comment by pronebird** [comment]
Review status: 5 of 6 files reviewed at latest revision, 2 unresolved discussions.

---

*[README.md, line 138 at r3](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/44#-L3csfsyGwc34ACnSB94:-L3dCOjk429X1DW80_m7:b-4z5sgp) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/aa80134d9e6bc1df3862d80caa26f067083d1d64/README.md#L138)):*
<details><summary><i>Previously, faern (Linus Färnstrand) wrote…</i></summary><blockquote>

Imagine someone like me cloning this repo and wanting to play with building and running the app. I don't even know what Babel is. I just want a series of commands to run in order to get the GUI running. Maybe I want to hack on the backend, but I don't know any JS/Babel/Node/Electron, I just want instructions that will help me get that part working.
</blockquote></details>

This is a reminder to all to us who work with the Javascript aspect of the app. If you don't have anything to do with this you should skip the bullet. There is a link to react-native project which is a clear indication of what's this is related to.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/44)*

**Comment by faern** [comment]
Reviewed 1 of 2 files at r5.
Review status: 5 of 6 files reviewed at latest revision, 2 unresolved discussions.

---

*[README.md, line 138 at r3](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44#-L3csfsyGwc34ACnSB94:-L3dDWNMEsOW9yegAZQF:b-yb0gt4) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/aa80134d9e6bc1df3862d80caa26f067083d1d64/README.md#L138)):*
<details><summary><i>Previously, pronebird (Andrei Mihailov) wrote…</i></summary><blockquote>

This is a reminder to all to us who work with the Javascript aspect of the app. If you don't have anything to do with this you should skip the bullet. There is a link to react-native project which is a clear indication of what's this is related to.
</blockquote></details>

Exactly my point! :) If that information is not relevant to me I don't want to have to read an entire issue to understand that. Can't we just explain here and now in one sentence that you don't need to care about this unless you are working on the JS parts and on react native?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44)*

**Comment by pronebird** [comment]
Review status: 5 of 6 files reviewed at latest revision, 2 unresolved discussions.

---

*[README.md, line 138 at r3](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/44#-L3csfsyGwc34ACnSB94:-L3dDyCgDXiL7b-Twxjk:bdm8rdk) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/aa80134d9e6bc1df3862d80caa26f067083d1d64/README.md#L138)):*
<details><summary><i>Previously, faern (Linus Färnstrand) wrote…</i></summary><blockquote>

Exactly my point! :) If that information is not relevant to me I don't want to have to read an entire issue to understand that. Can't we just explain here and now in one sentence that you don't need to care about this unless you are working on the JS parts and on react native?
</blockquote></details>

I proposed to describe the issue that Babel configuration under the "development" key is exclusively used by react-native due to the bug [ISSUE-URL]. Was that your point or am I missed something?

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/44)*

**Comment by pronebird** [comment]
Reviewed 1 of 2 files at r5.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/44)*

**Comment by faern** [comment]
Review status: 5 of 6 files reviewed at latest revision, 2 unresolved discussions.

---

*[README.md, line 138 at r3](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44#-L3csfsyGwc34ACnSB94:-L3dFGc73ULhhVujfrAW:bu0kd13) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/aa80134d9e6bc1df3862d80caa26f067083d1d64/README.md#L138)):*
<details><summary><i>Previously, pronebird (Andrei Mihailov) wrote…</i></summary><blockquote>

I proposed to describe the issue that Babel configuration under the "development" key is exclusively used by react-native due to the bug [ISSUE-URL]. Was that your point or am I missed something?
</blockquote></details>

I like this new version :lgtm: 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44)*

**Comment by eriklarko** [comment]
Reviewed 3 of 5 files at r1, 1 of 2 files at r5, 1 of 1 files at r6.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[app/app.android.js, line 26 at r6](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44#-L3gB3xMffw6E6_byi9Q:-L3gB3xMffw6E6_byi9R:b-emgc6l) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/27e4e18f8f3daa87bbda1f9a7f9eab74408f2204/app/app.android.js#L26)):*
> ```JavaScript
> 
> DeviceEventEmitter.addListener('com.mullvad.backend-info', function(e: Event) {
>   backend.init();
> ```

This logic differs from https://github.com/mullvad/mullvadvpn-app/blob/master/app/app.js#L26, perhaps we should extract that function and invoke it from here too?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44)*

**Comment by anderklander** [comment]
Review status: 5 of 6 files reviewed at latest revision, 1 unresolved discussion.

---

*[app/app.android.js, line 26 at r6](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44#-L3gB3xMffw6E6_byi9Q:-L3gCXXgnfcTDZ1xKGes:b-z8jupg) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/27e4e18f8f3daa87bbda1f9a7f9eab74408f2204/app/app.android.js#L26)):*
<details><summary><i>Previously, eriklarko (Erik Larkö) wrote…</i></summary><blockquote>

This logic differs from https://github.com/mullvad/mullvadvpn-app/blob/master/app/app.js#L26, perhaps we should extract that function and invoke it from here too?
</blockquote></details>

I have updated to use the same logic as app.js. I think there are several things that should/could be extracted or moved, but i feel like it comes in second hand, have not looked at or updated this file for a while since i assume most of it will be reworked when we have the possibility to actually render the app on android or have a backend. I should have removed stuff and commented … 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44)*

**Comment by eriklarko** [comment]
Review status: 5 of 6 files reviewed at latest revision, 1 unresolved discussion.

---

*[app/app.android.js, line 26 at r6](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44#-L3gB3xMffw6E6_byi9Q:-L3gcqtTyJM-etF6mqHV:b-caw7pv) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/27e4e18f8f3daa87bbda1f9a7f9eab74408f2204/app/app.android.js#L26)):*
<details><summary><i>Previously, anderklander wrote…</i></summary><blockquote>

I have updated to use the same logic as app.js. I think there are several things that should/could be extracted or moved, but i feel like it comes in second hand, have not looked at or updated this file for a while since i assume most of it will be reworked when we have the possibility to actually render the app on android or have a backend. I should have removed stuff and commented … 
</blockquote></details>

Alright, good point

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44)*

**Comment by pronebird** [comment]
Reviewed 1 of 1 files at r6, 1 of 1 files at r7.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/44)*

**Comment by eriklarko** [comment]
On linux I get `Error: Cannot find module 'browser-sync/lib/connect-utils'`, looking into it

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44#-:-L3h3sPPWuO3me99-U9T:bdlv6c8)*

**Comment by pronebird** [comment]
Btw: this is dev dependency only.

---

Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44#-:-L3h4muukbYzQYcbqJqW:bgl1hn6)*

**Comment by anderklander** [comment]
I also got that error at some point, but guessed that it was due to my fiddling with the modules and accidentally added a newer version of browser-sync, just guessed it was some issue with yarn or that i forgot to install after some change, tested to force browser sync version to 2.9.3 and it was ok again. Not an expert on yarn, but i guess the lock file would be the only possible difference?

---

Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44#-:-L3h73c6dahszF0QruCi:b7cf39f)*

**Comment by eriklarko** [comment]
I solved the missing module by changing `browser-sync/lib/connect-utils` to `browser-sync/dist/connect-utils`, but the hot reload doesn't work now. It could be that it is using the default babel env and need to be configured. This part of our code base is very hairy though, so it will take me some time to understand it all.

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44#-:-L3h9w0GvbHXKQscG5Y1:blxji9w)*

**Comment by pronebird** [comment]
I don't think it's related to babel at all. I looked through the NPM history and it seems that `lib` folder is no longer published on NPM. The last version that had `lib` is https://unpkg.com/browser-sync@2.21.0/

---

Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44#-:-L3hDBDPQUZqHNMaU4rf:buxah8i)*

**Comment by eriklarko** [comment]
the `lib` -> `dist` wasn't, but the hot reload errors were. They're fixed now though.

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44#-:-L3hEHhVgCJj0-oYwdOq:b-5q1d3d)*

**Comment by pronebird** [comment]
Yeah it seems like babel tried to compile the SVG as Javascript.

---

Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44#-:-L3hFV7qtHilqTyVEtpx:b9iiuof)*

**Comment by eriklarko** [comment]
*[a discussion](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44#-L3hNSvK34d4d29kDBX7:-L3hNSvLFHdMuCfVPKGm:b-otskb4) (no related file):*
I'm fine with this PR now

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44)*

**Comment by pronebird** [comment]
:lgtm: 

---

Reviewed 3 of 3 files at r8.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/44#-:-L3hNcsWgWx4r68iSvBG:bg7igkv)*
